### ISSUE TITLE
osmtracker for special purposes

### ISSUE THREAD

**Issue by jamescr** [nguillaumin/osmtracker-android#85]
Osmtracker is excellent. i created [a custom layout](https://github.com/jamescr/osmtracker-android/tree/layout_alajuela) for mapping a couple of things related with fire hydrants and local public transport. For me it's perfect but problems begin when I explain to another person, without technical knowledge, how to use the custom layout and how to upload data to OSM in a way I could find it latter.

I think a general approach to make osmtracker customizable to fit special purposes could be positive for the tool. For example a mechanism to download a custom layout, setting default values for the trace details and uploading the gpx to a specific server or service (not only OSM). 

The are other issues related ( #83 #84 #69 ) to this ideas but, I'll like to hear if this idea/proposal of enhacement makes sense.

**Comment by nguillaumin** [comment]
Yep, it makes sense.

**Comment by altaros97** [comment]
We are developing this Issue in two parts:  

A) Facilitate the download and installation of custom Layouts, creating a new activity in the settings menu. 
B) Upload traces to different servers, in addition to OSM.  

We are going to start solving the first part (A). I am going to open a new Issue where I will explain our development proposal.

**Comment by jsavage** [comment]
I would agree. Am looking for a custom layout for recording objects for openseamap

**Comment by zeliboba** [comment]
I totaly support idea to upload the tracks to a particular OSM server, e.g. to private or enterprise server.

**Comment by jamescr** [comment]
https://github.com/labexp/osmtracker-android/pull/270 added sharing GPX feature, a step forward uploading a GPX to a specific server.
